packages/recorder/src/index.ts

```diff
@@ -3,21 +3,53 @@ import {SessionManager} from './session_recording/sessions_manager';
 import {initTerminal} from './terminal_interaction';
 import {AppDependencies} from './types';
 
-const defaultDestination = process.env.DESTINATION;
-const useDestinationHostHeader = process.env.USE_DESTINATION_HOST_HEADER;
+import {Command} from 'commander';
+const program = new Command();
 
-if (!defaultDestination && useDestinationHostHeader !== 'true') {
-    console.log('Please provide a destination via the DESTINATION enviornment variable, or bypass this check by setting the USE_DESTINATION_HOST_HEADER bool environment variable');
+const runServerForDestination = async (host: string, port: string, deps: AppDependencies) => {
+    const app = initApp({
+        ...deps,
+        defaultDestination: host,
+    });
+
+    return new Promise<void>(r => {
+        app.listen(port, () => {
+            console.log(`http://localhost:${port}`);
+            setTimeout(() => {
+                r();
+            }, 50);
+        });
+    });
 }
 
-const sessionManager = new SessionManager();
-const deps: AppDependencies = {sessionManager, defaultDestination};
+program
+    .name('fc-record')
+    .description('CLI to record HTTP request sessions')
+    .version('0.0.1');
+
+// Usage: Destination host and local port pairs. A vertical bar is used to join the destination host and local port. The destinations are separated by spaces.
+program.command('record')
+    .description('Record requests')
+    .option('-d, --destinations [host|port...]', '')
+    .action(async (parsed: {destinations: string[]}, options) => {
```

Comment:
This file is the entrypoint for the record command. Here we're using a library `commander` to create a CLI program. We're declaring the command and saying that it takes a `--destinations` flag of this syntax:

```
record --destinations https://google.com|3000 https://twitter.com|3001
```

Which means: "Listen on port `3000`, and proxy any incoming requests to port `3000` to `google.com`. Then record all requests and responses into JSON files. It will also listen on `3001` to send to `twitter.com`. When the session summary is complete, it will output a timeline to a JSON file, of all requests/responses to be received by each host.